test/user.spec.js

```diff
@@ -0,0 +1,170 @@
+    };
+    request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(invalidPasswordLengthMessage);
+        expect(response.statusCode).to.equal(422);
+      });
+  });
+
+  it('should respond with 422 when password is badly formatted (not containing letters and numbers)', () => {
+    const password = chance.word({ length: 10 });
+    const errorUser = {
+      ...mockUser,
+      password
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(invalidPasswordMessage);
+        expect(response.statusCode).to.equal(422);
+      });
+  });
+
+  it('should respond with 422 when email is not @wolox.com.ar', () => {
+    const errorUser = {
+      ...mockUser,
+      email: chance.email({ domain: 'different.com.ar' })
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(invalidEmailDomain);
+        expect(response.statusCode).to.equal(422);
+      });
+  });
+
+  it('should respond with 400 when firstName is missing', () => {
+    const errorUser = {
+      ...mockUser,
+      firstName: null
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(missingRequiredFieldsMessage);
+        expect(response.body.message).to.include('firstName');
+        expect(response.statusCode).to.equal(400);
+      });
+  });
+
+  it('should respond with 400 when lastName is missing', () => {
+    const errorUser = {
+      ...mockUser,
+      lastName: null
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(missingRequiredFieldsMessage);
+        expect(response.body.message).to.include('lastName');
+        expect(response.statusCode).to.equal(400);
+      });
+  });
+
+  it('should respond with 400 when email is missing', () => {
+    const errorUser = {
+      ...mockUser,
+      email: null
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(missingRequiredFieldsMessage);
+        expect(response.body.message).to.include('email');
+        expect(response.statusCode).to.equal(400);
+      });
+  });
+
+  it('should respond with 400 when password is missing', () => {
+    const errorUser = {
+      ...mockUser,
+      password: null
+    };
+    return request
+      .post('/users')
+      .send(errorUser)
+      .then(response => {
+        expect(response.body.message).to.include(missingRequiredFieldsMessage);
+        expect(response.body.message).to.include('password');
+        expect(response.statusCode).to.equal(400);
+      });
+  });
+
+  it('should list all missing fields', () => {
```

Comment:
We normally name success tests like `'should success ...'` and `'should fail ... because...'`, i don't know if having  status codes in description of tests add something meaning